app-modules/contact/src/Rules/UniqueDomain.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+/*
+<COPYRIGHT>
+
+    Copyright © 2016-2024, Canyon GBS LLC. All rights reserved.
+
+    Aiding App™ is licensed under the Elastic License 2.0. For more details,
+    see <https://github.com/canyongbs/aidingapp/blob/main/LICENSE.>
+
+    Notice:
+
+    - You may not provide the software to third parties as a hosted or managed
+      service, where the service provides users with access to any substantial set of
+      the features or functionality of the software.
+    - You may not move, change, disable, or circumvent the license key functionality
+      in the software, and you may not remove or obscure any functionality in the
+      software that is protected by the license key.
+    - You may not alter, remove, or obscure any licensing, copyright, or other notices
+      of the licensor in the software. Any use of the licensor’s trademarks is subject
+      to applicable law.
+    - Canyon GBS LLC respects the intellectual property rights of others and expects the
+      same in return. Canyon GBS™ and Aiding App™ are registered trademarks of
+      Canyon GBS LLC, and we are committed to enforcing and protecting our trademarks
+      vigorously.
+    - The software solution, including services, infrastructure, and code, is offered as a
+      Software as a Service (SaaS) by Canyon GBS LLC.
+    - Use of this software implies agreement to the license terms and conditions as stated
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    <https://www.canyongbs.com> or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+namespace AidingApp\Contact\Rules;
+
+use Closure;
+use AidingApp\Contact\Models\Organization;
+use Illuminate\Contracts\Validation\ValidationRule;
+
+class UniqueDomain implements ValidationRule
+{
+    protected $ignoreId;
+
+    /**
+     * Create a new rule instance.
+     *
+     * @param  int|null  $ignoreId
+     */
+    public function __construct($ignoreId = null)
+    {
+        $this->ignoreId = $ignoreId;
+    }
+
+    /**
+     * Run the validation rule.
+     *
+     * @param  \Closure(string): \Illuminate\Translation\PotentiallyTranslatedString  $fail
+     */
+    public function validate(string $attribute, mixed $value, Closure $fail): void
+    {
+        $query = Organization::whereJsonContains('domains', [['domain' => $value]]);
+
+        if ($this->ignoreId) {
+            $query->where('id', '!=', $this->ignoreId);
+        }
+
+        $domainExists = $query->exists();
+
+        if ($domainExists) {
+            $fail($this->message());
+        }
```

Comment:
I think this may be able to be written a little cleaner, this is rough and I have not validated this is fully correct:

```suggestion
        if (
             Organization::whereJsonContains('domains', [['domain' => $value]])
                ->when(! empty($this->ignoreId), fn (Builder $query) => $query->where('id', '!=', $this->ignoreId))
                ->exists()
        ) {
            $fail($this->message());
        }
```